File: src/types/teacherClass.ts
export type RestrictionType = "preference" | "qualification";

export interface TeacherClassRestriction {
  id: number;
  teacherId: number;
  classId: number;
  restrictionType: RestrictionType;
}

export interface NewTeacherClassRestriction {
  teacherId: number;
  classId: number;
  restrictionType?: RestrictionType;
}

File: src-tauri/src/models/teacher_class.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TeacherClassRestriction {
    pub id: i64,
    pub teacher_id: i64,
    pub class_id: i64,
    pub restriction_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NewTeacherClassRestriction {
    pub teacher_id: i64,
    pub class_id: i64,
    pub restriction_type: Option<String>,
}
